prow/cmd/build/controller.go

```diff
@@ -307,6 +345,22 @@ func (c *controller) buildID(pj prowjobv1.ProwJob) (string, string, error) {
 	return id, url, nil
 }
 
+func (c *controller) terminateDupProwJobs(ctx string, namespace string) error {
+	jobs, err := c.getProwJobs(namespace)
+	if err != nil {
+		return err
+	}
+	return c.aborter.TerminateOlderPresubmitJobs(jobs, func(toCancel prowjobv1.ProwJob) error {
+		if c.config().Build.AllowCancellations {
+			err := c.deleteBuild(ctx, namespace, toCancel.GetName())
+			if err != nil && apierrors.IsNotFound(err) == false {
+				logrus.WithError(err).WithFields(pjutil.ProwJobFields(&toCancel)).Warn("Cannot delete build")
```

Comment:
Why not return err?

Reply:
I can return err. The idea was to keep doing the reconcile loop even though for some reason is not able to cleanup the duplicated jobs.

Reply:
If we never return an error let's update your interface definition to not return one, either one works.

Reply:
I think this remains untouched? If we don't want to return the error, let's change the function definition to not return one.

Reply:
And comment that cleanup functions should be best-effort

Reply:
I removed the error from the function signature and enhanced the comment.

Reply:
Please don't swallow the error.

`terminateDupProwJobs` and `TerminateOlderPresubmitJobs` both return errors. I see no reason the inner function shouldn't as well

Reply:
The error is not bubbled up because the abort process needs to continue to mark the jobs as aborted, whether or not an error occurred during cleanup. See below the `TerminateOlderPresubmitJobs` function. 

This was the original functionality from `Plank`, which I didn't want to change.

Reply:
Sure, but this should be something under `TerminateOlderPresubmitBuilds`' control that, not this function. 

AKA this function should return the error when `deleteBuild` fails, and `TerminateOlderPresubmitBuilds` can decide what to do with it (aka, log and continue)

Reply:
I changed the code to return the error to `TerminateOlderPresubmitBuilds`, which it will log it and continue.